### PR TITLE
[multiasic][sfputil]Fix the sftpuitlhelper on mutilasic platform issue

### PR DESCRIPTION
**Pull request by mlok-nokia**
#### Description
read_porttab_mappings() is not able to get the multiasic  port config.   Modified function read_porttab_mapings() to pass the asic_name to allow the get_port_config() to get the port info for each asic.  Also use the extend() and update() method to update the logical, logical_to_physical, phyiscal_to_logical lists.   Also add the unit test case test_read_all_porttab_mappings() .

#### Motivation and Context
This change is required to handle the mutliasic platform. Fixes https://github.com/Azure/sonic-buildimage/issues/10569

#### How Has This Been Tested?
The following output should be displayed without any error when using CLI command: sudo sfpulti show presence
```
admin@sonic:~$ sudo sfputil show presence
Port        Presence
----------  -----------
Ethernet0   Not present
Ethernet1   Not present
Ethernet2   Present
Ethernet3   Present
Ethernet4   Not present
Ethernet5   Not present
Ethernet6   Present
Ethernet7   Not present
Ethernet8   Not present
Ethernet9   Not present
Ethernet10  Not present
Ethernet11  Not present
Ethernet12  Not present
Ethernet13  Not present
Ethernet14  Not present
Ethernet15  Not present
Ethernet16  Not present
Ethernet17  Not present
Ethernet18  Not present
Ethernet19  Not present
Ethernet20  Not present
Ethernet21  Not present
Ethernet22  Not present
Ethernet23  Not present
Ethernet24  Present
Ethernet25  Present
Ethernet26  Present
Ethernet27  Not present
Ethernet28  Not present
Ethernet29  Not present
Ethernet30  Not present
Ethernet31  Not present
Ethernet32  Not present
Ethernet33  Not present
Ethernet34  Not present
Ethernet35  Not present
admin@sonic:~$ 
```

### CODE REVIEW

**Comment by mlok-nokia** [comment]
@rlhui @judyjoseph @prgeor  Please review this PR for  sfputilhelper on multiasic platform issue. Thanks!

**Comment by mlok-nokia** [comment]
@rlhui @judyjoseph @prgeor   Unit test case has been also added. Please review this PR.

File: sonic_platform_base/sonic_sfp/sfputilhelper.py [hunk 1]
```diff
@@ -45,15 +45,15 @@ class SfpUtilHelper(object):
     def __init__(self):
         pass
 
-    def read_porttab_mappings(self, porttabfile, asic_inst=0):
+    def read_porttab_mappings(self, porttabfile, asic_inst=0, asic_name=None):
```
Comment (judyjoseph) [inline sonic_platform_base/sonic_sfp/sfputilhelper.py:48]: Do we really need to add this additional parameter ?

File: sonic_platform_base/sonic_sfp/sfputilhelper.py [hunk 2]
```diff
@@ -45,15 +45,15 @@ class SfpUtilHelper(object):
...
         logical = []
         logical_to_physical = {}
         physical_to_logical = {}
         fp_port_index = 1
 
         (platform, hwsku) = device_info.get_platform_and_hwsku()
-        ports, _, _ = get_port_config(hwsku, platform)
-
+        ports, _, _ = get_port_config(hwsku, platform, asic_name=asic_name)
+
```
Comment (judyjoseph) [inline sonic_platform_base/sonic_sfp/sfputilhelper.py:56]: Additional whitespace

File: sonic_platform_base/sonic_sfp/sfputilhelper.py [hunk 3]
```diff
@@ -83,24 +83,25 @@ def read_porttab_mappings(self, porttabfile, asic_inst=0):
             # Mapping of logical port names available on a system to ASIC instance
             self.logical_to_asic[intf_name] = asic_inst
 
-        self.logical = logical
-        self.logical_to_physical = logical_to_physical
-        self.physical_to_logical = physical_to_logical
+        self.logical.extend(logical)
+        self.logical = list(set(self.logical))
```
Comment (judyjoseph) [inline sonic_platform_base/sonic_sfp/sfputilhelper.py:87]: Do we need this? as "logical" is anyways a list which we extend and its elements should be unique

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 1 above)
Comment (mlok-nokia) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:48]: Currently, the existing parameter "asic_inst" use 0 as default for non-asic instance.  But 0 is is a valid instance ID for multiasic case.  And this function could be used in many place with the 0 as non-asic instance. We cannot simply enhance it for the mutliasic case.   Using an additional parameter is a simple solution to  identify the non-asic and multiasic case.

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 3 above)
Comment (mlok-nokia) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:87]: The current logic of existing multiasic caller function read_all_porttab_mappings()  could parses the platform.json file number of times based on the number of asic.   The "logical" is not  unique.  and the 'extend' doesn't take care the duplicate element.

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 2 above)
Comment (mlok-nokia) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:56]: Fixed it.

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 3 above)
Comment (rlhui) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:87]: @judyjoseph  - reminder of this.

**Review by judyjoseph — DISMISSED** [review]

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 1 above)
Comment (judyjoseph) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:48]: Thinking again, @mlok-nokia I think we can avoid this parameter. Can we check if it is multi-asic and then pass asic_name to get_port_config API ?
``` 
ASIC_NAME_PREFIX = 'asic'

asic_name=none
if is_multi_asic:
    asic_name= ASIC_NAME_PREFIX++str(inst)
```

Comment (mlok-nokia) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py]: Fixed as suggested above

**Comment by prgeor** [comment]
@dgsudharsan can you review?

**Comment by prgeor** [comment]
@mlok-nokia how are the changes in bcmshell.py related to the sfputil issue?

File: sonic_platform_base/sonic_sfp/sfputilhelper.py [hunk 4]
```diff
@@ -83,24 +89,25 @@ def read_porttab_mappings(self, porttabfile, asic_inst=0):
...
+        self.logical_to_physical.update(logical_to_physical)
+        self.physical_to_logical.update(physical_to_logical)
 
         return None
 
 
     def read_all_porttab_mappings(self, platform_dir, num_asic_inst):
```
Comment (prgeor) [inline sonic_platform_base/sonic_sfp/sfputilhelper.py:100]: @mlok-nokia can you test this in non-multi-asic platform?

**Comment by mlok-nokia** [comment]
> @mlok-nokia how are the changes in bcmshell.py related to the sfputil issue?

This change has nothing to do with the sfputil.   During the build/UT and PEP8/flake8 verification, it complains the syntax as warning. I just fix them in the same commit.

File: sonic_platform_base/sonic_sfp/sfputilhelper.py (hunk 4 above)
Comment (mlok-nokia) [reply sonic_platform_base/sonic_sfp/sfputilhelper.py:100]: We don't have the non-multi-asic platform to test this change.  But there is an existing  UT test_read_port_mappings() for the it.

**Review by judyjoseph — APPROVED** [review]

**Pull request merged**